restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentSearchPageState.java

```diff
@@ -0,0 +1,41 @@
+package io.stargate.web.docsapi.dao;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.nio.ByteBuffer;
+import java.util.Base64;
+
+public class DocumentSearchPageState {
+  @JsonProperty("documentId")
+  String documentId;
+
+  @JsonProperty("internalPageState")
+  String internalPageState;
+
+  @JsonCreator
+  public DocumentSearchPageState(
+      @JsonProperty("documentId") final String documentId,
+      @JsonProperty("internalPageState") final String internalPageState) {
+    this.documentId = documentId;
+    this.internalPageState = internalPageState;
```

Comment:
```suggestion
    this.documentId = Strings.nullToEmpty(documentId);
    this.internalPageState = Strings.nullToEmpty(internalPageState);
```

Reply:
JSON should be able to encode `null` values, so why complicate the processing logic?

Reply:
I thought I had explained this in a comment somewhere on this PR< but to write it here as well:
JSON can encode null values, sure - but if Jackson is taking data such as
`{"documentId": "1", "internalPageState": null}`
And tries to turn that into a DocumentSearchPageState object, there is no way for it to know which constructor to use, as the `null` could be referring to a String or a ByteBuffer (there are two constructors in the class)

So I decided to represent the "missing" internal page state in JSON as the empty string - this technically does not have to be done for the `documentId`

Reply:
Thanks for the clarification - it does make sense.